spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -67,11 +69,15 @@ public boolean caseSensitive() {
   }
 
   public boolean localityEnabled() {
-    if (table.io() instanceof HadoopFileIO) {
-      HadoopInputFile file = (HadoopInputFile) table.io().newInputFile(table.location());
-      String scheme = file.getFileSystem().getScheme();
-      boolean defaultValue = LOCALITY_WHITELIST_FS.contains(scheme);
-      return PropertyUtil.propertyAsBoolean(readOptions, SparkReadOptions.LOCALITY, defaultValue);
+    if (table.io() instanceof HadoopFileIO || table.io() instanceof ResolvingFileIO) {
+      InputFile file = table.io().newInputFile(table.location());
+      if (file instanceof HadoopInputFile) {
+        String scheme = ((HadoopInputFile) file).getFileSystem().getScheme();
+        boolean defaultValue = LOCALITY_WHITELIST_FS.contains(scheme);
+        return PropertyUtil.propertyAsBoolean(readOptions, SparkReadOptions.LOCALITY, defaultValue);
+      } else {
+        return false;
```

Comment:
Nit: could table.io() be assigned to a variable?

Reply:
I am fine with the current way since `.io()` is quite short. Up to you Prashant.